CI/build-mudlet-for-windows.sh

```diff
@@ -150,7 +156,10 @@ export WITH_MAIN_BUILD_SYSTEM="NO"
 echo "Running qmake to make MAKEFILE ..."
 echo ""
 
-if [ "${MSYSTEM}" = "MINGW64" ]; then
+# TODO: This can be made better by just making a QMAKE and SPEC (or similar) variable
```

Comment:
Something like this then? (I configure `CC` and `CXX` in my scripts in the part which examines what the value of `MSYSTEM` is):
```bash
if [ "${CXX}" = "clang++" ]; then
  SPEC="win32-clang-g++"
else
  SPEC="win32-g++"
fi

# We do not use QtQuick so there is no need for those features:
QT_CONFIG=( \
  "CONFIG-=qml_debug" \
  "CONFIG-=qtquickcompiler" \
  )
if [ "${BUILD_CONFIG}" = "debug" ]; then
  QT_CONFIG+=( \
    "CONFIG+=debug" \
    "CONFIG+=separate_debug_info" \
    )
fi

echo "Checking for ccache..."
echo ""
if [ -f "${MINGW_INTERNAL_BASE_DIR}/bin/ccache" ]; then
  echo "  ... ccache has been found and it will try to be used for this build by"
  echo "passing extra argments to the qmake call."
  CCACHE_ARGS=( \
    "QMAKE_CC=ccache ${CC}" \
    "QMAKE_CXX=ccache ${CXX}" \
    )
else
  echo "  ... ccache has NOT been found!"
  unset CCACHE_ARGS
fi
echo ""

if [ "${QT_MAJOR_VERSION}" = "6" ]; then
  qmake6 "${CCACHE_ARGS[@]}" ../src/mudlet.pro -spec "${SPEC}" "${QT_CONFIG[@]}"
else
  qmake "${CCACHE_ARGS[@]}" ../src/mudlet.pro -spec "${SPEC}" "${QT_CONFIG[@]}"
fi
```